### PR TITLE
Prettify .js, .json files

### LINKED ISSUE #198: Use Prettier
[What is Prettier?](https://prettier.io/docs/en/index.html)
[Why Prettier?](https://prettier.io/docs/en/why-prettier.html)
😍

### PR DESCRIPTION
**Pull request by MarckK**
Pin Prettier v1.10.2 as dev dependency
Close #198
Signed-off-by: Kara de la Marck <karadelamarck@gmail.com>

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,129 +1,112 @@
-Changes by Version
-==================
+# Changes by Version
```
Comment (black-adder) [inline CHANGELOG.md]: can we skip .md files for now? This file post pretty looks incorrectly formatted.

Comment (MarckK) [reply CHANGELOG.md]: Yes. Also, @black-adder, if you have any formatting preferences, please let me know. I've just used Prettier's default ones, as I don't have strong preferences for formatting, other than it be done automatically.

**Comment by pavolloffay** [comment]
How did you format it? Is the format target run automatically during the build/commit? Is there a target to format it?

**Comment by MarckK** [comment]
Thus far, I've used Prettier's default formatting. This is just a start to incorporate Prettier in the project. Next steps would be setting it up so that Prettier will run against all relevant files that are staged (pre-commit).

**Comment by MarckK** [comment]
We could also use a script that runs Prettier as part of our CI checks, and that fails the build if relevant files differ from the Prettier output for that file.

**Comment by pavolloffay** [comment]
If we are forcing some style I would prefer to actually fail the build if the code is not formatted properly. Rather that precommit I like more npm/make target which can be run anytime.

**Comment by MarckK** [comment]
We can definitely have a format script, that is able to be run at anytime. The pre-commit hook is just another layer of reinforcement. Also, because this is opensource, the pre-commit hook makes it easier for people dropping in to make a PR (it removes the necessity of running the format script).

**Comment by MarckK** [comment]
@Pavol, I'll make another commit excluding the .md files from being autoformatted. Prettier also has [options](https://prettier.io/docs/en/options.html) for formatting, so is configurable. If there are style choices you feel strongly about, I should add them in now as well. :)

**Comment by yurishkuro** [comment]
@tiffon do you use Prettier in jaeger-ui?

**Comment by tiffon** [comment]
@yurishkuro Yes, we use prettier in Jaeger UI; it's great. I recommend setting the line length to 110. 

FWIW, Jaeger UI uses a precommit hook and does not validate in CI.

**Comment by yurishkuro** [comment]
I would also be in favor of pre-commit hook. Nothing is more annoying when the Travis build fails because `go fmt` wasn't run on a file.

@MarckK do you want to borrow the setup from jaeger-ui repo then?

**Comment by MarckK** [comment]
Yes. Sounds good 👍 :)

**Comment by MarckK** [comment]
Coverage is reduced as the number of lines has increased with the formatting done by Prettier (ratio of tests per line is reduced, so more untested lines).

**Comment by pavolloffay** [comment]
How does it work?
* is code style checked in the build?
* is formatting automatic? If not how to format the code?

**Comment by MarckK** [comment]
Hi @pavolloffay, `npm run format` will use prettier to format your local copy of the repo. 

Currently, this is using a pre-commit hook to prettify (to our specifications), so the formatting is done automatically before every commit. This makes it easier for others to contribute to the open source project (they don't have to remember to run format before committing).

There isn't a CI check for formatting, re Yuri's comment above: "I would also be in favor of pre-commit hook. Nothing is more annoying when the Travis build fails because go fmt wasn't run on a file."

File: Makefile [hunk 2]
```diff
@@ -14,7 +14,7 @@ test: build-node
 .PHONY: test-without-build
 test-without-build:
 	npm run flow
-	npm run lint
+	#npm run lint
```
Comment (pavolloffay) [inline Makefile]: this seems suspicious

Comment (MarckK) [reply Makefile]: Yeah, I should take out the commented out line.  The reason `npm run lint` is being taken out is that we have reconfigured Travis to lint on LTS versions of node.

**Comment by pavolloffay** [comment]
I have tested it locally with `v6.11.2` all worked well including precommit hook

@yurishkuro can we merge this?

**Comment by black-adder** [comment]
@tiffon can you take a look at the new dependencies in package.json and the commit hook? This looks fine to land but want some expert eyes.

**Comment by tiffon** [comment]
@black-adder, just seeing your comment, will look it over again tonight :+1:

**Comment by tiffon** [comment]
@black-adder - This looks great.

Thanks for the awesome work @MarckK!

**Pull request merged**